src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,207 @@
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+        
+        rightController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+
+        leftTargetSpeed = new SmartNumber("Edwin/Left Target Speed", 0);
+        rightTargetSpeed = new SmartNumber("Edwin/Right TargetSpeed", 0);
+
+        odometry = new DifferentialDriveOdometry(getRotation2d());
+        kinematics = new DifferentialDriveKinematics(TRACK_WIDTH);
+        navx = new AHRS(SPI.Port.kMXP);
+
+        setPose(new Pose2d());
+
+        field = new Field2d();
+        SmartDashboard.putData("Edwin/Field", field);
+    }
+
+    private void configMotors(CANSparkMax[] left, CANSparkMax[] right) {
+
+        leftEncoder.setVelocityConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION / 60);
+        leftEncoder.setPositionConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION);
+        rightEncoder.setVelocityConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION / 60);
+        rightEncoder.setPositionConversionFactor(Encoders.GEAR_DISTANCE_PER_ROTATION);
+
+        for (CANSparkMax motor : left) {
+            LEFT.configure(motor);
+        }
+
+        for (CANSparkMax motor : right) {
+            RIGHT.configure(motor);
+        }
+    }
+
+    /*********************
+     * ENCODER FUNCTIONS *
+     *********************/
+
+    // Distance
+    private double getLeftDistance() {
+        return leftEncoder.getPosition();
+    }
+
+    private double getRightDistance() {
+        return rightEncoder.getPosition();
+    }
+
+    // Velocity
+    private double getLeftVelocity() {
+        return leftEncoder.getVelocity();
+    }
+
+    private double getRightVelocity() {
+        return rightEncoder.getVelocity();
+    }
+
+    /***************
+     * ROBOT ANGLE *
+     ***************/
+
+    // Gets current Angle of the Robot as a Rotation2d (contiuous / not +-180)
+    public Rotation2d getRotation2d() {
+        return navx.getRotation2d();
+    }
+
+    /**********************
+     * ODOMETRY FUNCTIONS *
+     **********************/
+
+    public Pose2d getPose() {
+        return odometry.getPoseMeters();
+    }
+
+    @Override
+    public void setPose(Pose2d pose) {
+        leftEncoder.setPosition(0.);
+        rightEncoder.setPosition(0.);
+
+        odometry.resetPosition(pose, getRotation2d());
+    }
+
+    @Override
+    public DifferentialDriveKinematics getKinematics() {
+        return kinematics;
+    }
+
+    @Override
+    public TrajectoryConfig getTrajectoryConfig() {
+        return new TrajectoryConfig(
+            Motion.MAX_VELOCITY, 
+            Motion.MAX_ACCELERATION
+        ).setKinematics(kinematics);
+    }
+
+    /********************
+     * DRIVING COMMANDS *
+     ********************/
+
+    public void stop() {
+        driveVolts(0., 0.);
```

Comment:
this doesn't work becasue driveVolts will momentarily set the voltages to ZERO but then the next tick the pid loop that is always running in periodic will begin to PID itself back to the left and right target speeds. 


so i think just set the targets to zero and then call drivevolts zero or *just* set the targets to zero and let periodic handle it or get rid of the method.